crates/ruff_linter/src/rules/flake8_simplify/rules/open_file_with_context_handler.rs

```diff
@@ -114,6 +114,30 @@ fn match_exit_stack(semantic: &SemanticModel) -> bool {
     false
 }
 
+/// Return `true` if the current expression is nested in a call to one of the
+/// unittest context manager methods: `cls.enterClassContext()`,
+/// `self.enterContext()`, or `self.enterAsyncContext()`.
+fn match_unittest_context_methods(semantic: &SemanticModel) -> bool {
+    let Some(expr) = semantic.current_expression_parent() else {
+        return false;
+    };
+    let Expr::Call(ast::ExprCall { func, .. }) = expr else {
+        return false;
+    };
+    let Expr::Attribute(ast::ExprAttribute { attr, value, .. }) = func.as_ref() else {
+        return false;
+    };
+    if attr != "enterClassContext" && attr != "enterContext" && attr != "enterAsyncContext" {
```

Comment:
```suggestion
    if !matches!(attr, "enterClassContext" | "enterContext" | "enterAsyncContext") {
```

But also, isn't this check redundant since it's also checked on the last line?

Reply:
Aye, that it is. Removing.